include/deal.II/lac/trilinos_vector.h

```diff
@@ -20,36 +20,172 @@
 #include <deal.II/base/config.h>
 
 #ifdef DEAL_II_WITH_TRILINOS
-
 #  include <deal.II/base/subscriptor.h>
 #  include <deal.II/base/index_set.h>
 #  include <deal.II/base/utilities.h>
+#  include <deal.II/base/mpi.h>
 #  include <deal.II/lac/exceptions.h>
 #  include <deal.II/lac/vector.h>
-#  include <deal.II/lac/trilinos_vector_base.h>
 #  include <deal.II/lac/vector_type_traits.h>
 
+#  include <vector>
+#  include <utility>
+#  include <memory>
+
 DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#  include "Epetra_ConfigDefs.h"
+#  ifdef DEAL_II_WITH_MPI // only if MPI is installed
+#    include "mpi.h"
+#    include "Epetra_MpiComm.h"
+#  else
+#    include "Epetra_SerialComm.h"
+#  endif
+#  include "Epetra_FEVector.h"
 #  include "Epetra_Map.h"
 #  include "Epetra_LocalMap.h"
 DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 
-#include <memory>
-
 DEAL_II_NAMESPACE_OPEN
 
 
-// forward declaration
-template <typename> class Vector;
-
 /**
  * @addtogroup TrilinosWrappers
  * @{
  */
+
+/**
+ * A namespace in which wrapper classes for Trilinos objects reside.
+ *
+ * @ingroup TrilinosWrappers
+ */
 namespace TrilinosWrappers
 {
   class SparseMatrix;
 
+  /**
+   * @cond internal
+   */
+
+  /**
+   * A namespace for internal implementation details of the TrilinosWrapper
+   * members.
+   *
+   * @ingroup TrilinosWrappers
+   */
+  namespace internal
+  {
+    /**
+     * Declare type for container size.
+     */
+    typedef dealii::types::global_dof_index size_type;
+
+    /**
+     * This class implements a wrapper for accessing the Trilinos vector in
+     * the same way as we access deal.II objects: it is initialized with a
+     * vector and an element within it, and has a conversion operatorto
```

Comment:
Actually, it looks like we dropped the space after each `operator` in the documentation: please go back and fix this by searching for `operator[a-z]` or equivalent.